graph-http/src/intoresponse.rs

```diff
@@ -299,8 +326,30 @@ impl<'a, T> IntoResponseAsync<'a, T> {
         let headers = response.headers().clone();
         let status = response.status();
         let url = GraphUrl::from(response.url());
-        let json = response.json().await.map_err(GraphFailure::from)?;
-        Ok(GraphResponse::new(url, json, status, headers))
+        let mut json: U = response.json().await.map_err(GraphFailure::from)?;
+        let mut values: Vec<V> = vec![];
+
+        if self.client.client.lock().follow_next_links {
```

Comment:
Similar to how you added the method for setting the next link in the RequestClient trait we should add a method to get that value and call it her instead. I try to keep all of the places where we are calling `lock()` in only a few files so that we know where to find them and can debug them easier. `IntoResponse` shouldn't have any place that is calling the lock() method. I don't remember if there are places in there where its being done but if there is that is something that a ticket should get created for to fix it.

Adding a method with something like:
```rust
if self.client.should_follow_next_links()
```
or
```rust
if self.client.get_follow_next_links()
```
or
```rust
if self.client.following_next_links()
```

something like that.